apps/condo/domains/meter/schema/RegisterMetersReadingsService.js

```diff
@@ -0,0 +1,260 @@
+                                property: { connect: { id: property.id } },
+                                unitType: reading.addressMeta.unitType,
+                                unitName: reading.addressMeta.unitName,
+                                accountNumber: reading.accountNumber,
+                                number: reading.meterNumber,
+                                resource: { connect: reading.meterResource },
+                                numberOfTariffs: reading.meterMeta.numberOfTariffs,
+                                place: reading.meterMeta.place,
+                                verificationDate: toISO(reading.meterMeta.verificationDate),
+                                nextVerificationDate: toISO(reading.meterMeta.nextVerificationDate),
+                                installationDate: toISO(reading.meterMeta.installationDate),
+                                commissioningDate: toISO(reading.meterMeta.commissioningDate),
+                                sealingDate: toISO(reading.meterMeta.sealingDate),
+                                controlReadingsDate: reading.meterMeta.controlReadingsDate ? toISO(reading.meterMeta.controlReadingsDate) : dayjs().toISOString(),
+                            })
+                            meterId = createdMeter.id
+                        } catch (e) {
+                            resultRows.push(e)
+                            continue
+                        }
+                    }
+
+                    try {
+                        const values = ['value1', 'value2', 'value3', 'value4'].reduce((result, currentValue) => {
+                            const value = reading[currentValue]
+                            const normalizedValue = normalizeMeterValue(value)
+
+                            if (!validateMeterValue(normalizedValue)) {
+                                throw new GQLError(ERRORS.INVALID_METER_VALUE(currentValue, value), context)
+                            }
+
+                            return {
+                                ...result,
+                                [currentValue]: normalizedValue,
+                            }
+                        }, {})
+                        const createdMeterReading = await MeterReading.create(context, {
+                            dv,
+                            sender,
+                            meter: { connect: { id: meterId } },
+                            source: { connect: { id: IMPORT_CONDO_METER_READING_SOURCE_ID } },
+                            date: toISO(reading.date),
+                            ...values,
+                        })
+
+                        resultRows.push(createdMeterReading)
+                    } catch (e) {
+                        resultRows.push(e)
+                    }
+                }
+
+                const metersReadings = await find('MeterReading', { id_in: resultRows.map((row) => get(row, 'id')).filter(Boolean) })
```

Comment:
This will not guarantee you'll find a result...And that's the big problem. When we'll introduce DB replicas, there'll be no guarantee that your data will be here. Because there's replication lag. You should use data returned by create

Reply:
This is not the only mutation that finds the data just created. It looks like the problem you point out is out of scope for this PR.